### ISSUE TITLE
callback.reset_learning_rate doesn't works [upd: 19:47] type: python

### ISSUE THREAD

**Issue by paantya** [dmlc/xgboost#4709]
ups
editing... wait

upd(19:47):
Hi everyone!
i change learning_rate, and it doesn't work!(

```
import xgboost as xgb
xgb.__version__
```
>`'0.90'`

```
best_params = {"alpha": 1.0, 
               "base_score": 0, 
               "booster": "gbtree", 
               "colsample_bytree": 0.5, 
#                "disable_default_eval_metric": 1,
#                "eta": 0.5,
               "gamma": 0.75, 
               "grow_policy":"depthwise", 
               "lambda": 2.5, 
               "max_depth": 6, 
               "min_child_weight": 2,
               "objective": "reg:linear", 
               "seed": 1, 
               "subsample": 0.8, 
               "tree_method": "approx", 
               "verbosity": 1}
params = best_params

def cv_lerning_rate(my_chande_eta, verbose_eval=5,num_boost_round = 10):
    # feature = shap_feature_zero[:-2]
    evals = [(dtrain, 'tr'), (dtest, 'te')]
    print('params',params)
    print('num_boost_round',num_boost_round)
    print('verbose_eval',verbose_eval)
    print('my_chande_eta',my_chande_eta)
    xgb.train(params, dtrain, 
              num_boost_round=num_boost_round,
              evals=evals,
              verbose_eval=verbose_eval,
              learning_rates=my_chande_eta,
              callbacks  = [xgb.callback.reset_learning_rate(my_chande_eta)]
             )
```

1. The following launches should be different, but they are the same.

- a

```
exp_list = [ 0.1*2**i for i in range(10)]
cv_lerning_rate(exp_list, verbose_eval=1, num_boost_round=10)
```

```
> params {'booster': 'gbtree', 'min_child_weight': 2, 'tree_method': 'approx', 'objective': 'reg:linear', 'seed': 1, 'grow_policy': 'depthwise', 'max_depth': 6, 'base_score': 0, 'alpha': 1.0, 'subsample': 0.8, 'colsample_bytree': 0.5, 'gamma': 0.75, 'verbosity': 1, 'lambda': 2.5}
> num_boost_round 10
> verbose_eval 1
> my_chande_eta [0.1, 0.2, 0.4, 0.8, 1.6, 3.2, 6.4, 12.8, 25.6, 51.2]
> [16:01:20] WARNING: src/objective/regression_obj.cu:148: reg:linear is now deprecated in favor of reg:squarederror.
> [0]	tr-rmse:2.96025	te-rmse:3.60808
> [1]	tr-rmse:2.88991	te-rmse:3.55977
> [2]	tr-rmse:2.8304	te-rmse:3.51701
> [3]	tr-rmse:2.77981	te-rmse:3.48807
> [4]	tr-rmse:2.72521	te-rmse:3.45098
> [5]	tr-rmse:2.67641	te-rmse:3.41924
> [6]	tr-rmse:2.63638	te-rmse:3.38831
> [7]	tr-rmse:2.59527	te-rmse:3.35894
> [8]	tr-rmse:2.56036	te-rmse:3.33637
> [9]	tr-rmse:2.53378	te-rmse:3.3191

```
- b

```
exp_list = [ 0.1 for i in range(10)]
cv_lerning_rate(exp_list, verbose_eval=1, num_boost_round=10)
```

```
> params {'booster': 'gbtree', 'min_child_weight': 2, 'tree_method': 'approx', 'objective': 'reg:linear', 'seed': 1, 'grow_policy': 'depthwise', 'max_depth': 6, 'base_score': 0, 'alpha': 1.0, 'subsample': 0.8, 'colsample_bytree': 0.5, 'gamma': 0.75, 'verbosity': 1, 'lambda': 2.5}
> num_boost_round 10
> verbose_eval 1
> my_chande_eta [0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1]
> [15:57:07] WARNING: src/objective/regression_obj.cu:148: reg:linear is now deprecated in favor of reg:squarederror.
> [0]	tr-rmse:2.96025	te-rmse:3.60808
> [1]	tr-rmse:2.88991	te-rmse:3.55977
> [2]	tr-rmse:2.8304	te-rmse:3.51701
> [3]	tr-rmse:2.77981	te-rmse:3.48807
> [4]	tr-rmse:2.72521	te-rmse:3.45098
> [5]	tr-rmse:2.67641	te-rmse:3.41924
> [6]	tr-rmse:2.63638	te-rmse:3.38831
> [7]	tr-rmse:2.59527	te-rmse:3.35894
> [8]	tr-rmse:2.56036	te-rmse:3.33637
> [9]	tr-rmse:2.53378	te-rmse:3.3191
```

2. **Dosn't work:**

> NOTE: the initial learning rate will still take in-effect on first iteration.

The first and second runs should have the same value at the first iteration, but they are different. In addition, the second and third launch should have different values ​​(since the first iterations are guaranteed to be different, according to the documentation), but they are suspiciously the same

- a
```
params['eta'] = 0.5
exp_list = [ 0.5 for i in range(10)]
cv_lerning_rate(exp_list, verbose_eval=1, num_boost_round=10)

```

```
> params {'booster': 'gbtree', 'min_child_weight': 2, 'tree_method': 'approx', 'objective': 'reg:linear', 'seed': 1, 'grow_policy': 'depthwise', 'max_depth': 6, 'eta': 0.5, 'base_score': 0, 'alpha': 1.0, 'subsample': 0.8, 'colsample_bytree': 0.5, 'gamma': 0.75, 'verbosity': 1, 'lambda': 2.5}
> num_boost_round 10
> verbose_eval 1
> my_chande_eta [0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5]
> [16:04:53] WARNING: src/objective/regression_obj.cu:148: reg:linear is now deprecated in favor of reg:squarederror.
> [0]	tr-rmse:2.70482	te-rmse:3.45844
> [1]	tr-rmse:2.54976	te-rmse:3.3464
> [2]	tr-rmse:2.47275	te-rmse:3.28536
> [3]	tr-rmse:2.41417	te-rmse:3.24429
> [4]	tr-rmse:2.3543	te-rmse:3.20143
> [5]	tr-rmse:2.32154	te-rmse:3.18285
> [6]	tr-rmse:2.30218	te-rmse:3.17717
> [7]	tr-rmse:2.28183	te-rmse:3.17314
> [8]	tr-rmse:2.26009	te-rmse:3.16311
> [9]	tr-rmse:2.23464	te-rmse:3.16288
```

- b
```
params['eta'] = 0.5
exp_list = [ 0.1 for i in range(10)]
cv_lerning_rate(exp_list, verbose_eval=1, num_boost_round=10)
```

```
> params {'booster': 'gbtree', 'min_child_weight': 2, 'tree_method': 'approx', 'objective': 'reg:linear', 'seed': 1, 'grow_policy': 'depthwise', 'max_depth': 6, 'eta': 0.5, 'base_score': 0, 'alpha': 1.0, 'subsample': 0.8, 'colsample_bytree': 0.5, 'gamma': 0.75, 'verbosity': 1, 'lambda': 2.5}
> num_boost_round 10
> verbose_eval 1
> my_chande_eta [0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1]
> [16:05:27] WARNING: src/objective/regression_obj.cu:148: reg:linear is now deprecated in favor of reg:squarederror.
> [0]	tr-rmse:2.96025	te-rmse:3.60808
> [1]	tr-rmse:2.88991	te-rmse:3.55977
> [2]	tr-rmse:2.8304	te-rmse:3.51701
> [3]	tr-rmse:2.77981	te-rmse:3.48807
> [4]	tr-rmse:2.72521	te-rmse:3.45098
> [5]	tr-rmse:2.67641	te-rmse:3.41924
> [6]	tr-rmse:2.63638	te-rmse:3.38831
> [7]	tr-rmse:2.59527	te-rmse:3.35894
> [8]	tr-rmse:2.56036	te-rmse:3.33637
> [9]	tr-rmse:2.53378	te-rmse:3.3191
```

- c
```
best_params = {"alpha": 1.0, 
               "base_score": 0, 
               "booster": "gbtree", 
               "colsample_bytree": 0.5, 
#                "disable_default_eval_metric": 1,
#                "eta": 0.5,
               "gamma": 0.75, 
               "grow_policy":"depthwise", 
               "lambda": 2.5, 
               "max_depth": 6, 
               "min_child_weight": 2,
               "objective": "reg:linear", 
               "seed": 1, 
               "subsample": 0.8, 
               "tree_method": "approx", 
               "verbosity": 1}
params = best_params
exp_list = [ 0.1 for i in range(10)]
cv_lerning_rate(exp_list, verbose_eval=1, num_boost_round=10)
```

```
> params {'booster': 'gbtree', 'min_child_weight': 2, 'tree_method': 'approx', 'objective': 'reg:linear', 'seed': 1, 'grow_policy': 'depthwise', 'max_depth': 6, 'base_score': 0, 'alpha': 1.0, 'subsample': 0.8, 'colsample_bytree': 0.5, 'gamma': 0.75, 'verbosity': 1, 'lambda': 2.5}
> num_boost_round 10
> verbose_eval 1
> my_chande_eta [0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1]
> [16:08:18] WARNING: src/objective/regression_obj.cu:148: reg:linear is now deprecated in favor of reg:squarederror.
> [0]	tr-rmse:2.96025	te-rmse:3.60808
> [1]	tr-rmse:2.88991	te-rmse:3.55977
> [2]	tr-rmse:2.8304	te-rmse:3.51701
> [3]	tr-rmse:2.77981	te-rmse:3.48807
> [4]	tr-rmse:2.72521	te-rmse:3.45098
> [5]	tr-rmse:2.67641	te-rmse:3.41924
> [6]	tr-rmse:2.63638	te-rmse:3.38831
> [7]	tr-rmse:2.59527	te-rmse:3.35894
> [8]	tr-rmse:2.56036	te-rmse:3.33637
> [9]	tr-rmse:2.53378	te-rmse:3.3191
```

**Comment by paantya** [comment]
Help pls!

**Comment by paantya** [comment]
it is not this error (https://github.com/dmlc/xgboost/pull/5155)

**Comment by trivialfis** [comment]
@paantya I think your issue is fixed in #5155

**Comment by trivialfis** [comment]
There's nightly build you can try, see the note section of installation doc.
